tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -198,19 +212,56 @@ class RouteConfigMatcher {
     }
     return pathParts;
   }
+}
 
-  /**
-   * In case of parameter projection failure (cannot create pathname), it throws
-   * a RangeError.
-   */
-  matchByParams(params: RouteParams): PositiveMatch {
-    const pathParts = this.reprojectPathByParams(this.pathFragments, params);
+class ConcreteRouteConfigMatcher extends RouteConfigMatcher {
+  constructor(readonly definition: ConcreteRouteDef) {
+    super(definition);
+  }
+}
+
+class StaticRedirectionRouteConfigMatcher extends RouteConfigMatcher {
+  private readonly redirectionFragments: PathFragment[];
+
+  constructor(readonly definition: RedirectionRouteDef) {
```

Comment:
Can we drop the `readonly`?  The `definition` does not need to be stored as a property

Reply:
It is read in the downstream `RouteConfigMatchers`.

Reply:
While this specific `definition is not read below, having some congruent interface in all `RouteConfigMatcher` definitely is something I was going for so I would like to keep it as is.